source/slang/slang-parameter-binding.cpp

```diff
@@ -1065,81 +1075,87 @@ static void addExplicitParameterBindings_GLSL(
     // the index/offset/etc.
     //
 
-    TypeLayout::ResourceInfo* resInfo = nullptr;
-    TypeLayout::ResourceInfo* foundResInfo = nullptr;
-
-    LayoutSemanticInfo semanticInfo;
-    semanticInfo.index = 0;
-    semanticInfo.space = 0;
+    const int kMaxResCount = 2;
+    const int kSubpassResInfo = 1;
+    const int kResInfo = 0;
 
-    LayoutSemanticInfo subpassSemanticInfo;
-    bool foundBinding = false;
-    bool foundSubpass = false;
+    TypeLayout::ResourceInfo* foundResInfo = nullptr;
+    struct ResAndSemanticInfo
+    {
+        TypeLayout::ResourceInfo* resInfo = nullptr;
+        LayoutSemanticInfo semanticInfo;
```

Comment:
You can just defined it as
LayoutSemanticInfo semanticInfo {};

so you don't need the constructor.

Reply:
The code before my changes does not zero `LayoutSemanticInfo::kind`, `LayoutSemanticInfo semanticInfo {};` would zero everything.

Reply:
Well, by looking at previous code, I think they should zero `kind` as well, though the `kind` will be assigned anyway.